### ISSUE TITLE
dartdevk Crash on type bounds issue

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#43303]
The following test crashes dartdevk
```dart
import "../../Utils/expect.dart";

class C<X> {
  test(X x) {
    var v = {x, 1, 2};
    Expect.isTrue(v is Set<Object>);
    Expect.isFalse(v is Set<int>);
  }
}

main() {
  C<int>().test(42);
}
```
Output is
```
/=========================================================================================================================================\
| dartdevk-strong-linux-release-chrome:co19/LanguageFeatures/Set-literals/type_inference_A20_t01 is new and failed (Crash, expected Pass) |
\=========================================================================================================================================/

--- Command "dartdevk" (took 117ms):
DART_CONFIGURATION=ReleaseX64 CHROME_PATH=/b/s/w/ir/cache/builder/sdk/third_party/browsers/chrome/chrome/google-chrome out/ReleaseX64/dart-sdk/bin/dartdevc --enable-experiment=non-nullable --enable-experiment=non-nullable --sound-null-safety --ignore-unrecognized-flags --no-summarize -o /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/generated_compilations/dartdevk-checked-sdk/tests_co19_src_LanguageFeatures_Set-literals_type_inference_A20_t01/type_inference_A20_t01.js /b/s/w/ir/cache/builder/sdk/tests/co19/src/LanguageFeatures/Set-literals/type_inference_A20_t01.dart -s /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/gen/utils/dartdevc/pkg_sound/async_helper.dill=async_helper -s /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/gen/utils/dartdevc/pkg_sound/expect.dill=expect -s /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/gen/utils/dartdevc/pkg_sound/js.dill=js -s /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/gen/utils/dartdevc/pkg_sound/meta.dill=meta (working directory: /b/s/w/ir/cache/builder/sdk/tests/co19/src/LanguageFeatures/Set-literals)

exit code:
253

stdout:
We're sorry, you've found a bug in our compiler.
You can report this bug at:
    https://github.com/dart-lang/sdk/issues/labels/web-dev-compiler
Please include the information below in your report, along with
any other information that may help us track it down. Thanks!
-------------------- %< --------------------
    dartdevc -k arguments: --enable-experiment=non-nullable --enable-experiment=non-nullable --sound-null-safety --ignore-unrecognized-flags --no-summarize -o /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/generated_compilations/dartdevk-checked-sdk/tests_co19_src_LanguageFeatures_Set-literals_type_inference_A20_t01/type_inference_A20_t01.js /b/s/w/ir/cache/builder/sdk/tests/co19/src/LanguageFeatures/Set-literals/type_inference_A20_t01.dart -s /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/gen/utils/dartdevc/pkg_sound/async_helper.dill=async_helper -s /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/gen/utils/dartdevc/pkg_sound/expect.dill=expect -s /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/gen/utils/dartdevc/pkg_sound/js.dill=js -s /b/s/w/ir/cache/builder/sdk/out/ReleaseX64/gen/utils/dartdevc/pkg_sound/meta.dill=meta
    dart --version: 2.10.0-edge.a7cb764aa457cedf07f7a0e73b86d34bcc6d2e5a (be) (Wed Sep 2 12:12:52 2020 +0000) on "linux_x64"

Unsupported operation: Undetermined Nullability
#0      ProgramCompiler._emitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:2662:7)
#1      ProgramCompiler.visitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:2584:7)
#2      InterfaceType.accept (package:kernel/ast.dart:7554:42)
#3      MappedListIterable.elementAt (dart:_internal/iterable.dart:417:31)
#4      MappedListIterable.elementAt (dart:_internal/iterable.dart:417:40)
#5      ListIterator.moveNext (dart:_internal/iterable.dart:343:26)
#6      InstantiatorGeneratorVisitor.splayNodes (package:dev_compiler/src/js_ast/template.dart:220:23)
#7      InstantiatorGeneratorVisitor.handleCallOrNew.<anonymous closure> (package:dev_compiler/src/js_ast/template.dart:583:22)
#8      Template.instantiate (package:dev_compiler/src/js_ast/template.dart:109:26)
#9      JsBuilder.call (package:dev_compiler/src/js_ast/builder.dart:214:21)
#10     ProgramCompiler._emitGenericClassType (package:dev_compiler/src/kernel/compiler.dart:2743:15)
#11     ProgramCompiler._emitInterfaceType (package:dev_compiler/src/kernel/compiler.dart:2634:17)
#12     ProgramCompiler.visitSetLiteral (package:dev_compiler/src/kernel/compiler.dart:5451:21)
#13     SetLiteral.accept (package:kernel/ast.dart:5389:44)
#14     ProgramCompiler._visitExpression (package:dev_compiler/src/kernel/compiler.dart:3412:20)
#15     ProgramCompiler._visitInitializer (package:dev_compiler/src/kernel/compiler.dart:1538:11)
#16     ProgramCompiler.visitVariableDeclaration (package:dev_compiler/src/kernel/compiler.dart:4069:13)
#17     VariableDeclaration.accept (package:kernel/ast.dart:6987:43)
#18     ProgramCompiler._visitStatement (package:dev_compiler/src/kernel/compiler.dart:3340:20)
#19     MappedListIterable.elementAt (dart:_internal/iterable.dart:417:31)
#20     ListIterator.moveNext (dart:_internal/iterable.dart:343:26)
#21     new List.from (dart:core-patch/array_patch.dart:38:29)
#22     new List.of (dart:core-patch/array_patch.dart:68:17)
#23     ListIterable.toList (dart:_internal/iterable.dart:211:44)
#24     ProgramCompiler.visitBlock (package:dev_compiler/src/kernel/compiler.dart:3506:62)
#25     Block.accept (package:kernel/ast.dart:5873:43)
#26     ProgramCompiler._visitStatement (package:dev_compiler/src/kernel/compiler.dart:3340:20)
#27     ProgramCompiler._emitFunctionScopedBody (package:dev_compiler/src/kernel/compiler.dart:3351:18)
#28     ProgramCompiler._emitSyncFunctionBody.<anonymous closure> (package:dev_compiler/src/kernel/compiler.dart:3159:17)
#29     ProgramCompiler._withLetScope (package:dev_compiler/src/kernel/compiler.dart:2098:25)
#30     ProgramCompiler._withCurrentFunction (package:dev_compiler/src/kernel/compiler.dart:3193:18)
#31     ProgramCompiler._emitSyncFunctionBody (package:dev_compiler/src/kernel/compiler.dart:3155:17)
#32     ProgramCompiler._emitFunction (package:dev_compiler/src/kernel/compiler.dart:2958:11)
#33     ProgramCompiler._emitMethodDeclaration (package:dev_compiler/src/kernel/compiler.dart:1717:12)
#34     ProgramCompiler._emitClassMethods (package:dev_compiler/src/kernel/compiler.dart:1658:23)
#35     ProgramCompiler._emitClassDeclaration (package:dev_compiler/src/kernel/compiler.dart:559:21)
#36     ProgramCompiler._emitClass (package:dev_compiler/src/kernel/compiler.dart:501:21)
#37     List.forEach (dart:core-patch/growable_array.dart:313:8)
#38     ProgramCompiler._emitLibrary (package:dev_compiler/src/kernel/compiler.dart:450:23)
#39     ProgramCompiler._emitLibrary (package:dev_compiler/src/kernel/compiler.dart)
#40     List.forEach (dart:core-patch/growable_array.dart:313:8)
#41     ProgramCompiler.emitModule (package:dev_compiler/src/kernel/compiler.dart:341:15)
#42     _compile (package:dev_compiler/src/kernel/command.dart:406:27)
<asynchronous suspension>
#43     compile (package:dev_compiler/src/kernel/command.dart:46:18)
#44     compile (package:dev_compiler/src/compiler/shared_command.dart:437:10)
#45     runBatch (file:///b/s/w/ir/cache/builder/sdk/pkg/dev_compiler/bin/dartdevc.dart:92:22)
<asynchronous suspension>
#46     main (file:///b/s/w/ir/cache/builder/sdk/pkg/dev_compiler/bin/dartdevc.dart:30:11)
#47     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:297:32)
#48     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:168:12)

--- Re-run this test:
python tools/test.py -n dartdevk-strong-linux-release-chrome co19/LanguageFeatures/Set-literals/type_inference_A20_t01
```
Please see also #43280

**Comment by nshahan** [comment]
I'm adding this to our tracking list since this is hitting a throw I added as part of the null safety support.

**Comment by nshahan** [comment]
This test uses a set literal but the same problem exists with set, list, and map literals.

**Comment by nshahan** [comment]
Fixed by a CFE change d3114bac63b8ec22e8dfee4bb91b3f4c8c1fab59.
